docs/rfcs/header-component-api.md

```diff
@@ -0,0 +1,418 @@
+
+![Navigation secondary link](images/header-component-api/navigation-secondary-link.png)
+
+###### Props
+
+-   `href` _string_
+    -   navigation location when `Navigation.SecondaryLink` is clicked
+-   `label` _string_
+    -   display label of the `Navigation.SecondaryLink`
+
+###### Example
+
+```jsx
+<Navigation.SecondaryLink
+    label="World news"
+    href="https://www.theguardian.com/world"
+/>
+```
+
+##### `<Navigation.SecondaryLinksSupplementalLink />`
+
+A visually prominent navigational link that appears in the `Navigation.SecondaryLinks` component.
+
+![Navigation secondary links supplemental link](images/header-component-api/navigation-secondary-links-supplemental-link.png)
+
+###### Example
+
+```jsx
+<Navigation.SecondaryLinksSupplementalLink
+    label="Masterclasses"
+    href="https://www.theguardian.com/guardian-masterclasses"
+/>
+```
+
+###### Props
+
+-   `href` _string_
+    -   navigation location when `Navigation.SecondaryLinksSupplementalLink` is clicked
+-   `label` _string_
+    -   display label of the `Navigation.SecondaryLinksSupplementalLink`
+
+## Specialised headers
+
+### `<EditorialHeader isSignedIn showSubscriberMessage edition />`
+
+A special header layout for editorial pages. The logo and supporter messages are hard-coded.
+
+![Editorial header](images/header-component-api/editorial-header.png)
+
+#### Differences to generic header
+
+##### Brandbar
+
+-   Links to the left, floated right
+-   Editions dropdown to the right
+-   My account / sign in
+    -   Links hierarchy
+    -   Primary link – always visible
+    -   Secondary links – visible at wider breakpoints
+    -   Editions menu
+    -   Is this a concept we want to reflect in the API?
+-   Subscribe / Contribute links
+-   `<SubscribeMessage />`?
+    -   Hardcoded in Source?
+    -   Currently inserted by automat
+
+#### Example
+
+```jsx
+<EditorialHeader isSignedIn showMessage>
+    <BrandBar>
+        <BrandBar.Links>
+            <BrandBar.Link href icon={<Icon />} label until from />
+            <BrandBar.LinkList icon label until from>
+                <BrandBar.Link href label />
+                <BrandBar.Link href label />
+            </BrandBar.LinkList>
+            <BrandBar.Link href icon={<Icon />} label until from />
+        </BrandBar.Links>
+        <EditorialHeader.EditionSwitch onChange>
+            <EditorialHeader.EditionSwitchItem href label />
+            <EditorialHeader.EditionSwitchItem href label selected />
+            <EditorialHeader.EditionSwitchItem href label />
+        </EditorialHeader.EditionSwitch>
+    </BrandBar>
+    <Navigation />
+    <Subnav />
+</EditorialHeader>
+```
+
+#### Props
+
+-   `isSignedIn` _boolean_
+    -   whether to show the support buttons
+-   `showSubscriberMessage` _boolean_
+    -   whether to show the Thank You message
+-   `edition` _string_ `'uk'` | `'au'` | `'us'` | `'international'`
+
+#### `<EditorialHeader.EditionSwitch />`
```

Comment:
As a first pass, should we hardcode this component into the `EditorialHeader`, rather than exposing it as part of the API? It seems unlikely that consumers would want to customise it or change it once implemented.

Reply:
we'd need to make sure it included all the tracking links etc, which feels like the wrong type of thing to keep in source?

Reply:
i wonder if it's a viable rule to exclude any business logic from our component, and always require state derived from business logic to be passed in?

Reply:
This last separation makes sense. I think it will be easier to use this component if the business logic is held locally (e.g., in DCR) and the thing that matters to Source, and what Source brings to the table, is presentation